iOSFeedbackExamples/EmailViewController.swift

```diff
@@ -7,13 +7,112 @@
 //
 
 import UIKit
+import MessageUI
 
-class EmailViewController: UIViewController {
+class EmailViewController: UIViewController, MFMailComposeViewControllerDelegate {
 
     override func viewDidLoad() {
         super.viewDidLoad()
 
         navigationItem.title = "Email"
     }
-
+    @IBAction func sendEmailAction(_ sender: UIButton) {
+        if MFMailComposeViewController.canSendMail() {
+            // 注意这个实例要写在if block里，
+            // 否则无法发送邮件时会出现两次提示弹窗（一次是系统的）
+            let mailComposeViewController = configuredMailComposeViewController()
+            self.present(mailComposeViewController, animated: true, completion: nil)
+        } else {
+            self.showSendMailErrorAlert()
+        }
+    }
+    // delegate
+    func mailComposeController(_ controller: MFMailComposeViewController,
+                               didFinishWith result: MFMailComposeResult,
+                               error: Error?) {
+        self.dismiss(animated: true, completion: nil)
+        switch result {
+        case .cancelled:    print("canceld");   break;
+        case .saved:        print("saved");     break;
+        case .sent:         print("sent");      break;
+        case .failed:       print("failed");    break;
+        }
+        self.dismiss(animated: true, completion: nil)
+    }
+    
+    func configuredMailComposeViewController() -> MFMailComposeViewController {
+        
+        let mailComposeVC = MFMailComposeViewController()
+        mailComposeVC.mailComposeDelegate = self
+        // 获取系统版本号
+        let systemVersion = UIDevice.current.systemVersion
+        // 获取设备的型号
+        let deviceModel = UIDevice.current.model
+        // 设置收件地址
+        mailComposeVC.setToRecipients(["1038875231@qq.com"])
+        // 设置邮件主题
+        mailComposeVC.setSubject("提交反馈")
+        // 设置邮件正文
+        mailComposeVC.setMessageBody("请填写您的建议, 我们将尽快处理:\n\n\n\n\n\n系统版本：\(systemVersion)\n设备型号：\(deviceModel)", isHTML: false)
+        
+        return mailComposeVC
+    }
+    // 用户没有设置自己邮箱时
+    func showSendMailErrorAlert() {
+        
+        let sendMailErrorAlert = UIAlertController(title: "无法发送邮件",
+                                                   message: "您的设备尚未设置邮箱，请在“邮件”应用中设置后再尝试发送。",
+                                                   preferredStyle: .alert)
+        sendMailErrorAlert.addAction(UIAlertAction(title: "取消", style: .cancel) {
+            _ in
+            _ = self.navigationController?.popToRootViewController(animated: true)
+        })
+        sendMailErrorAlert.addAction(UIAlertAction(title: "去设置", style: .default) {
+            _ in
+            self.gotoSettingPage()
+        })
```

Comment:
In RateUsViewController, `UIAlertAction` has a variable to hold it, code style!!